x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/ClusterDeprecationChecks.java

```diff
@@ -189,4 +192,29 @@ static DeprecationIssue checkTemplatesWithMultipleTypes(ClusterState state) {
             + " define multiple types and so will cause errors when used in index creation"
             );
     }
+
+    static DeprecationIssue checkFollowedSystemIndices(ClusterState state) {
+        Set<String> systemIndexFollowers = new HashSet<>();
+        for (ObjectObjectCursor<String, IndexMetadata> indexEntry : state.metadata().getIndices()) {
+            final IndexMetadata indexMetadata = indexEntry.value;
+            final Map<String, String> ccrMetadata = indexMetadata.getCustomData(CCR.CCR_CUSTOM_METADATA_KEY);
+            if (ccrMetadata == null) {
+                continue;
+            }
+            if (indexMetadata.isSystem()) {
+                systemIndexFollowers.add(indexEntry.key);
+            }
+        }
+
+        if (systemIndexFollowers.isEmpty()) {
+            return null;
+        }
+
+        return new DeprecationIssue(DeprecationIssue.Level.WARNING,
+            "Some follower indices follow remote system indices",
+            "https://www.elastic.co/guide/en/elasticsearch/reference/master/breaking-changes-8.0.html",
```

Comment:
Where should I add the deprecation note?

Reply:
I believe it should go to `docs/reference/release-notes/7.14.asciidoc`, you may need to add a reference from `release-notes.asciidoc` to it.